src/chunk.c

```diff
@@ -3054,6 +3063,54 @@ chunk_update_status(FormData_chunk *form)
 							   CurrentMemoryContext) > 0;
 }
 
+/* status update is done in 2 steps.
+ * 1. RowShare lock to read the status.
+ * 2. if status != proposed new status
+ *      update status using RowExclusiveLock
+ * All callers who want to update chunk status should call this function so that locks
```

Comment:
I don't really get it why we need to take the table-level `RowShare` and `RowExclusiveLock` locks here. First of all, they don't conflict with each other, or with themselves.

I do see how `LockTupleExclusive` as a row-level `FOR UPDATE` lock can solve the problem, but it is not mentioned in this comment whatsoever.

Reply:
One is at the table level , the other is for the tuple.